test/common/quic/envoy_quic_h3_fuzz_test.cc

```diff
@@ -0,0 +1,768 @@
+#include "source/common/quic/envoy_quic_alarm_factory.h"
+#include "source/common/quic/envoy_quic_connection_helper.h"
+#include "source/common/quic/envoy_quic_dispatcher.h"
+#include "source/common/quic/envoy_quic_server_connection.h"
+#include "source/common/quic/envoy_quic_server_session.h"
+#include "source/common/quic/envoy_quic_utils.h"
+#include "source/common/quic/platform/quiche_logging_impl.h"
+#include "source/common/quic/server_codec_impl.h"
+
+#include "test/common/quic/envoy_quic_h3_fuzz.pb.h"
+#include "test/common/quic/test_proof_source.h"
+#include "test/common/quic/test_utils.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/network/mocks.h"
+
+#include "quiche/common/quiche_data_writer.h"
+#include "quiche/quic/core/crypto/null_decrypter.h"
+#include "quiche/quic/core/crypto/null_encrypter.h"
+#include "quiche/quic/core/deterministic_connection_id_generator.h"
+#include "quiche/quic/core/quic_crypto_server_stream.h"
+#include "quiche/quic/core/quic_dispatcher.h"
+#include "quiche/quic/core/tls_server_handshaker.h"
+#include "quiche/quic/test_tools/quic_test_utils.h"
+
+namespace Envoy {
+namespace Quic {
+
+const size_t kMaxNumPackets = 10;
+
+const size_t kMaxQuicPacketSize = 1460;
+static char quic_packets[kMaxNumPackets][kMaxQuicPacketSize];
+static size_t quic_packet_sizes[kMaxNumPackets];
```

Comment:
Can the global variables be avoided?

Reply:
Yes, I packed them into the classes that use them, or on the stack, where possible.